plugins/thunderbolt/meson.build

```diff
@@ -1,8 +1,15 @@
 cargs = ['-DG_LOG_DOMAIN="FuPluginThunderbolt"']
+tbt_deps = [plugin_deps, gudev]
+
+if get_option('enable-dell')
+  tbt_deps += libsmbios_c
+  tbt_deps += efivar
+endif
 
 shared_module('fu_plugin_thunderbolt',
   sources : [
     'fu-plugin-thunderbolt.c',
+    'fu-dell-common.c',
```

Comment:
You're right, if `get_option('enable-dell') `is false this shouldn't be included.
The problem with doing hwid check is that it's dependent upon a variety of SMBIOS strings that could potentially be rebranded on an OEM'ed system.  HWID would cover >99% of the cases though.